task3Daniv/codeanalyzer.cpp

```diff
@@ -0,0 +1,152 @@
+#include "stdafx.h"
+#include "codeanalyzer.h"
+
+#include <regex>
+#include <thread>
+#include <chrono>
+#include <fstream>
+#include <sstream>
+#include <iostream>
+#include <algorithm>
+
+CodeAnalyzer::CodeAnalyzer(const std::experimental::filesystem::path& path)
+	: mFile{path},
+	mFileName{ path.filename().string() }
+{
+}
+
+std::string CodeAnalyzer::getResult()
+{
+	std::string str((std::istreambuf_iterator<char>(mFile)),
+		std::istreambuf_iterator<char>());
+	mCode = str;
+
+	mCode.erase(std::remove(mCode.begin(), mCode.end(), '\t'), mCode.end());
```

Comment:
This approach is unoptimized because **mCode** is array-like container. **std::remove()** algorithm moves elements toward the end of this array which costs much if the array is large. Better use **std::regex_replace()**